include/RAJA/pattern/kernel/Lambda.hpp

```diff
@@ -64,6 +64,18 @@ struct Lambda : internal::Statement<camp::nil> {
   const static camp::idx_t loop_body_index = BodyIdx;
 };
 
+//Base case
```

Comment:
This is really great work.

Here are some suggestions:

It would be great if tLambda got merged into Lambda...
template<camp::idx_t BodyIndex, typename ... Args>

Where the current Lambda implementation is a specialization with zero Args.

It would also be great if you could say
Lambda<0, Seg<0>, Param<0>, Seg<1>>
Which would pass to a lambda the following arguments: value of segment 0, parameter 0, value of segment 1.
(Clearly we would need to create a Seg<camp::idx_t> type)

Or,
Lambda<0, Seg<0>, Offset<0>>
Which would pass the arguments: the value of segment 0, the offset of segment 0.
This would obviate the need for the ForICount, and eliminate the need for defining a param to hold the ICount value since it would just be passed in by argument.

Reply:
I agree with @ajkunen 's first comment. I assume you would want to do that too, but wanted to get some early feedback before doing more surgery to the code.

@ajkunen second comment is similar to the one I made above.

We should probably discuss @ajkunen 's third comment with others to see what they think. While the API may be leaner, it may be too opaque for some folks.

Reply:
With next iteration, I've made Lambda variadic so the new variation is just an extension. I'm currently working on getting something like Lambda<0, Seg<0>, Param<0>, Seg<1>> to work. I think I'm not too far.